marconi/test/Spec/Utxo.hs

```diff
@@ -22,73 +26,180 @@ import Cardano.Api qualified as C
     let (targetAddresses :: Maybe TargetAddresses ) = addressesFromTxOuts txOuts
     let (utxos :: [Utxo.Utxo]) = Utxo.getUtxos targetAddresses t
     case targetAddresses of
-        Nothing         ->  (length utxos) === (length txOuts)
+        Nothing         ->  length utxos === length txOuts
         Just targets    ->
             ( nub
-              . catMaybes
-              . fmap (\x -> addressAnyToShelley (x ^. Utxo.utxoAddress))
+              . mapMaybe (\x -> storableQueryToShelleyAddr (x ^. Utxo.address))
               $ utxos) === (nub . toList $ targets)
 
-genBlockNo :: Gen C.BlockNo
-genBlockNo = C.BlockNo <$> Gen.word64 Range.constantBounded
-
-genEvents :: Gen Utxo.UtxoEvent
-genEvents = do
-    slotNo <- CGen.genSlotNo
-    blockNo  <- genBlockNo
-    txs <- Gen.list (Range.linear 2 5)(CGen.genTx C.ShelleyEra)
-    pure . fromJust $ Utxo.getUtxoEvents Nothing slotNo blockNo txs
-
 utxoStorageTest :: Property
 utxoStorageTest = property $ do
-    depth <- forAll $ Gen.int (Range.linear 1  5) -- force DB writes
-    ndx <- liftIO $ Utxo.open ":memory:" (Utxo.Depth depth)
-    events <- forAll $ Gen.list (Range.linear 1 10) genEvents
-    let (rows :: [Utxo.UtxoRow]) = nub . concatMap Utxo.toRows $ events
-        (addresses :: [C.AddressAny]) = nub . fmap (\r -> r ^. Utxo.utxoRowUtxo . Utxo.utxoAddress ) $ rows
-    ix <- liftIO $ Ix.insertL (events) ndx
-    rowsFromStore <- liftIO $ (concat . catMaybes) <$> forM addresses (\a -> Utxo.queryPlusVolatile ix a  )
-    (null rowsFromStore) === False
-    let (tid :: Utxo.UtxoRow -> C.AddressAny) = (\r -> r ^. Utxo.utxoRowUtxo . Utxo.utxoAddress)
-    (all (`elem` (tid <$> rows)) (tid <$> rowsFromStore)) === True
+  events <- forAll $ Gen.list (Range.linear 2 7) genEvents
+  (storedEvents :: [StorableEvent Utxo.UtxoHandle]) <-
+    (liftIO . Utxo.open ":memory:") (Utxo.Depth  10) -- no DB writes
+     >>= liftIO . Storable.insertMany events
+     >>= liftIO . Storable.getEvents
+  Hedgehog.assert (equivalentLists storedEvents events)
 
 addressesFromTxOuts :: [C.TxOut C.CtxTx CurrentEra] -> Maybe TargetAddresses
-addressesFromTxOuts [(C.TxOut addressInEra _ _ _)]
+addressesFromTxOuts [C.TxOut addressInEra _ _ _]
     = nonEmpty
     . nub
-    . catMaybes
-    . (fmap (addressAnyToShelley . Utxo.toAddr) )
+    . mapMaybe (addressAnyToShelley . Utxo.toAddr)
     $ [addressInEra]
 addressesFromTxOuts _ = Nothing
 
+storableQueryToShelleyAddr :: StorableQuery Utxo.UtxoHandle -> Maybe (C.Address C.ShelleyAddr)
+storableQueryToShelleyAddr (Utxo.UtxoAddress (C.AddressShelley a ) ) = Just a
+storableQueryToShelleyAddr _                                         = Nothing
+
 addressAnyToShelley :: C.AddressAny -> Maybe (C.Address C.ShelleyAddr)
-addressAnyToShelley  (C.AddressShelley a )=Just a
-addressAnyToShelley  _                     = Nothing
+addressAnyToShelley  (C.AddressShelley a) = Just a
+addressAnyToShelley  _                    = Nothing
```

Comment:
Some property tests for `resumeFromStorage` as well.

Some examples would be:
* returns at least the ChainPointAtGenesis point
* it should return the last point of the data indexed on disk